x-pack/plugins/apm/public/components/shared/charts/instances_latency_distribution_chart/index.tsx

```diff
@@ -51,6 +61,43 @@ export function InstancesLatencyDistributionChart({
   const maxLatency = Math.max(...items.map((item) => item.latency ?? 0));
   const latencyFormatter = getDurationFormatter(maxLatency);
 
+  const tooltip: TooltipProps = {
+    type: TooltipType.Follow,
+    snap: false,
+    customTooltip: (props: TooltipInfo) => (
+      <CustomTooltip {...props} latencyFormatter={latencyFormatter} />
+    ),
+  };
+
+  /**
+   * Handle click events on the items.
+   *
+   * Due to how we handle filtering by using the kuery bar, it's difficult to
+   * modify existing queries. If you have an existing query in the bar, this will
+   * wipe it out. This is ok for now, since we probably will be replacing this
+   * interaction with something nicer in a future release.
+   *
+   * The event object has an array two items for each point, one of which has
+   * the serviceNodeName, so we flatten the list and get the items we need to
+   * form a query.
+   */
+  const handleElementClick: ElementClickListener = (event) => {
+    const serviceNodeNamesQuery = event
+      .flat()
+      .flatMap((value) => (value as GeometryValue).datum?.serviceNodeName)
+      .filter((serviceNodeName) => !!serviceNodeName)
+      .map((serviceNodeName) => `${SERVICE_NODE_NAME}:"${serviceNodeName}"`)
+      .join(' OR ');
+
+    urlHelpers.push(history, { query: { kuery: serviceNodeNamesQuery } });
+  };
+
+  // With a linear scale, if all the instances have similar throughput (or if
+  // there's just a single instance) they'll show along the origin. Make sure
+  // the x-axis domain always starts at 0 and goes to double the max of the items
```

Comment:
Is it still 2x the max throughput?

```suggestion
  // the x-axis domain goes from [0, maxThroughput]
```